### PR TITLE
Feature/cache topics navigation

### PR DESCRIPTION
**Pull request by AWitcherONS**
### What

add cache to taxonomy add topics navigation

### How to review

Code complies to current standards
and test work

parameters in  run,sh ..
export ENABLE_NAVIGATION="Y"
export TOPICS_SERVICE_URL="http://localhost:25300" 

if ENABLE_NAVIGATION="N"  then taxonomy will come from zebedee

export DEV_ENVIRONMENT="Y" will add logging for cache 

### Who can review

anyone but me

### CODE REVIEW

**Review by DavidSubiros — COMMENTED** [review]
LGTM but there is a conflict with develop branch. 
- Is there any test I can do in a running system?
- If I run `make test` I observe teh following error:
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.948 s
[INFO] Finished at: 2023-01-16T10:20:49Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:testCompile (default-testCompile) on project babbage: Compilation failure
[ERROR] /Users/david/workspace/ons/babbage/src/test/java/com/github/onsdigital/babbage/content/client/ContentClientCacheTest.java:[40,64] cannot find symbol
[ERROR]   symbol:   method getContent(java.lang.String,java.util.HashMap<java.lang.String,java.lang.String[]>)
[ERROR]   location: variable contentClientCache_mock of type com.github.onsdigital.babbage.content.client.ContentClientCache
```

File: run.sh [hunk 1]
```diff
@@ -22,6 +22,8 @@ export DEV_ENVIRONMENT="Y"
 export RELOAD_TEMPLATES="Y"
 export TEMPLATES_DIR=src/main/web/templates/handlebars
 export ENABLE_COVID19_FEATURE=false
+export ENABLE_NAVIGATION="Y"
+export TOPICS_SERVICE_URL="http://localhost:25300"
```
Comment (lindenmckenzie) [inline run.sh:26]: I can see by default we're bypassing the router, would we do this in prod?

**Review by lindenmckenzie — COMMENTED** [review]
See comments - happy to discuss

File: run.sh (hunk 1 above)
Comment (AWitcherONS) [reply run.sh:26]: See dp-configs merged march 24th need to update staging/prod when ready to deploy

File: src/main/java/com/github/onsdigital/babbage/configuration/Babbage.java [hunk 2]
```diff
@@ -51,106 +51,122 @@ static Babbage getInstance() {
...
         maxCacheEntries = defaultIfBlank(getNumberValue(MAX_OBJECT_SIZE), 3000);
+        maxCacheObjectSize = defaultIfBlank(getNumberValue(MAX_CACHE_ENTRIES), 50000);
         maxHighchartsServerConnections = defaultIfBlank(getNumberValue("HIGHCHARTS_EXPORT_MAX_CONNECTION"), 50);
         maxResultsPerPage = 250;
         maxVisiblePaginatorLink = 5;
         publishCacheTimeout = 60 * 60;
         redirectSecret = getValueOrDefault(REDIRECT_SECRET_KEY, "secret");
         resultsPerPage = 10;
         searchResponseCacheTime = 5;
+        serviceAuthToken = getValueOrDefault(SERVICE_AUTH_TOKEN, " ahyofaem2ieVie6eipaX6ietigh1oeM0Aa1aiyaebiemiodaiJah0eenuchei1ai");
```
Comment (lindenmckenzie) [inline src/main/java/com/github/onsdigital/babbage/configuration/Babbage.java:89]: should there be a blank space at the beginning of this?

File: src/main/java/com/github/onsdigital/babbage/content/client/ContentClientCache.java [hunk 3]
```diff
@@ -33,13 +29,11 @@
  * Using Apache http client with connection pooling.
  */
 public class ContentClientCache {
-
+    private static final String BearerPrefix         = "Bearer ";
```
Comment (lindenmckenzie) [inline src/main/java/com/github/onsdigital/babbage/content/client/ContentClientCache.java:32]: Why is there a large gap here?

**Review by lindenmckenzie — COMMENTED** [review]
Couple of very minor comments.

**Review by lindenmckenzie — APPROVED** [review]

**Pull request merged**